reposerver/repository/repository.go

```diff
@@ -1125,6 +1131,14 @@ func helmTemplate(appPath string, repoRoot string, env *v1alpha1.Env, q *apiclie
 		return nil, fmt.Errorf("failed parsing dependencies: %v", err)
 	}
 
+	// We do a sanity check here to give a nicer error message in case any of the Helm repositories are not permitted by
+	// the AppProject which the application is a part of
+	for _, repo := range q.Repos {
+		if !isSourcePermitted(repo.Repo, q.ProjectSourceRepos) {
+			return nil, status.Errorf(codes.PermissionDenied, "helm repo %s is not permitted in project '%s'", repo.Repo, q.ProjectName)
+		}
+	}
```

Comment:
Should we create the list of repos that are not permitted and return in a single error message rather than returning error on the first repo that is not permitted?